### DISCUSSION TITLE
livewire query with limit() show less data in datalist text box

### DISCUSSION THREAD

**Discussion by rizkyprasetya** [livewire/livewire#5067] [1. Help] [answered]
Hello all,

php artisan livewire:make contactSelect

The contact-select.blade.php
```
<div>
    <label for="listContact">Select contact:</label>
    <input wire:model="optionContact" type="text" list="listContact" class="form-control">
    <datalist  name="" id="listContact">
        @foreach ($contacts as $contact)
        <option value="{{$contact->name}}">{{$contact->name}}</option>
        @endforeach
    </datalist>
</div>
```
the ContactSelect..php
```

<?php
namespace App\Http\Livewire;
use App\Models\Contact;
use Livewire\Component;

class ContactSelect extends Component
{
    public $optionContact = '';
    public function render()
    {
        return view(
            'livewire.contact-select',
            ['contacts' => Contact::where('name', 'like', "%$this->optionContact%")
                ->limit(5)
                ->get()]
        );
    }
}
```
when I try to change the value in the text input box repeatedly with different value (add some char and then delete/backspace some, and then add some char again), sometimes the datalist shows/return less than the right amount of data, i.e. some data didn't show up.  

This behavior didn't show up when the limit(5) line is deleted/commented, the datalist show the expected amount of data (5, or exactly the number  of data exist in table)

My goal is  to reduce traffic and server load.

Updated:
just found out this happens only with firefox, currently installed version 101.0.1.

**Comment by Martleby** [comment]
My guess is this is a DOM diffing issue. Try putting wire:key on your options.

HTH
Martin.

**Reply by rizkyprasetya** [reply]
Sorry for the late update, I found  out the problem. Firefox is rather picky about the URL. I started laravel in development environment with minimum option which defaulting to 127.0.0.1, but my .env is localhost. When I start laravel with --host localhost, firefox behaves accordingly.

**Comment by rizkyprasetya** [comment, marked as answer]
Start laravel with the same URL as in the .env APP_URL, for example if your APP_URL is localhost the start laravel with "php artisan serve --host localhost"
